pkg/apis/pipeline/v1alpha1/verificationpolicy_validation.go

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2022 The Tekton Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/tektoncd/pipeline/pkg/apis/config"
+	"github.com/tektoncd/pipeline/pkg/apis/validate"
+	"knative.dev/pkg/apis"
+)
+
+var _ apis.Validatable = (*VerificationPolicy)(nil)
+
+// Validate VerificationPolicy
+func (v *VerificationPolicy) Validate(ctx context.Context) (errs *apis.FieldError) {
+	cfg := config.FromContextOrDefaults(ctx)
+	if cfg.FeatureFlags.ResourceVerificationMode != config.EnforceResourceVerificationMode && cfg.FeatureFlags.ResourceVerificationMode != config.WarnResourceVerificationMode {
+		errs = errs.Also(apis.ErrGeneric(fmt.Sprintf("invalid feature flag %s, must be one of (%s, %s)", cfg.FeatureFlags.ResourceVerificationMode, config.EnforceResourceVerificationMode, config.WarnResourceVerificationMode)))
+		return errs
+	}
+	errs = errs.Also(validate.ObjectMetadata(v.GetObjectMeta()).ViaField("metadata"))
+	errs = errs.Also(v.Spec.Validate(ctx))
+	return errs
+}
+
+// Validate VerificationPolicySpec
+func (vs *VerificationPolicySpec) Validate(ctx context.Context) (errs *apis.FieldError) {
```

Comment:
Since this is an exported function, we might want a test for this explicitly as well. I think you test it via Validation of verification policy.

Reply:
yes we test it from `func (v *VerificationPolicy) Validate(ctx context.Context)`, it should cover this func and we don't need to test again?

Reply:
If this function actually needs to be exported then I think so since something else could also call it later. But if this function is only expected to be called by `VerificationPolicy`'s `Validate` function then how about not exporting it?That way, it won't require tests either.

Reply:
oh I got your point. I think `Spec.Validate` is usually exported, it is also the same for others resources in tekton. 
e.g. https://github.com/tektoncd/pipeline/blob/c3c07aa358b67a0982e0d3fd391d9cbb7ac31f05/pkg/apis/pipeline/v1alpha1/run_validation.go#L48

Reply:
though we don't have to test it directly